### PR TITLE
Rename CheckStyleTask to CheckstyleAntTask. #1108

### PR DESCRIPTION
**Pull request by mkordas**

### CODE REVIEW

**Comment by romani** [comment]
http://checkstyle.sourceforge.net/anttask.html
all is good except for one thing:

1) changes to resource file:
src/main/resources/checkstyletask.properties --> src/main/resources/checkstyle_ant_task.properties

that will braking compatibility, example of what user need to do: 
https://github.com/sevntu-checkstyle/checkstyle-samples/commit/59e20d4639553da3254e1d5a28e0e9e3860de78e

First of all I think usage of "_" in names old style, usage of "-" is better to read and that is trend in naming now. If you agree please to rename by "git mv" to keep git-history of the file.
Second, ..... the reason of such refactoring was a desire to separate Java files to make structure a bit more clear in modular. But for resources it is not a problem, to my mind. So we could keep compatibility with previous version by keeping name of that file the same. What do you think ?

2)
in HTML page there is config line : 
```
<project name="foo" ...
         xmlns:cs="antlib:com.puppycrawl.tools.checkstyle">
```

Am not very experienced ANT user , Should it be changed too ? looks like here package of  Task is specified.

**Comment by romani** [comment]
1) 
> src/main/resources/checkstyletask.properties

content of this file has to be changed. Please do testing yourself by sevntu-checkstyle/checkstyle-samples , right now it is failing.
```
BUILD FAILED
checkstyle-test/tmp/checkstyle-samples/ant-project/build.xml:26: taskdef class com.puppycrawl.tools.checkstyle.CheckStyleTask cannot be found

```

2) Please provide comment to the issue to describe users how to do migration of their configs.

**Comment by mkordas** [comment]
1) Fixed
2) I decided to keep backwards compatibility and not change name of `src/main/resources/checkstyletask.properties`. I agree that it should be named `checkstyle-ant-task`, but maybe we should do that during 7.0 release? We can keep track of such deferred changes using milestones.

**Comment by romani** [comment]
1) >  I agree that it should be named checkstyle-ant-task, but maybe we should do that during 7.0 release? We can keep track of such deferred changes using milestones.

I do not care about name of that file it does not make a problem for me for now.

2) Please do testing yourself by sevntu-checkstyle/checkstyle-samples and confirm that it does not fail

3) 
> We can keep track of such deferred changes using milestones.

I do not think that is required , as each minor release in CS is potential braking compatibility
such update is ok for 6.8.

**Comment by romani** [comment]
@mkordas ,  .... just in case you did not notice my last post.

**Comment by mkordas** [comment]
I've noticed - I'm working on it right now :)

**Comment by mkordas** [comment]
1) OK, renamed.
2) Testing done, everything works.
3) OK

**Comment by romani** [comment]
some misunderstanding is happening:

1) no  need to rename property file as it will reduce amount of compatibility brake. Issue was about refactoring in java level to make Checkstyle internals and integration(ant) a bit more clear.

2) in a issue a description for resolving upgrade from 6.7 to 6.8 we need to summarize all changes that you did for our xdoc as there were examples of different usage in ANT.

if smth still not clear lets resolve misunderstanding in chat.

**Comment by mkordas** [comment]
1) I understood _I do not care about name_ that you do not care if I changed that name, but this was misunderstanding, my bad. 

2) I think we can have full backwards compatibility, as docs do not mention using the class with task directly - so no notes needed.

Can you review PR now? It was tested with sevntu-checkstyle/checkstyle-samples.

**Comment by romani** [comment]
@mkordas ,

I worried about following config in out HTML site - http://checkstyle.sourceforge.net/anttask.html#Installation
```
<project name="foo" ...
         xmlns:cs="antlib:com.puppycrawl.tools.checkstyle">
```
behaviour is described there - http://ant.apache.org/manual/Types/antlib.html#antlibnamespace

we have file checkstyle/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/antlib.xml
that end up in result jar at following path: 
       checkstyle-6.7-all.jar/com/puppycrawl/tools/checkstyle/antlib.xml 
(fyi: checkstyle-6.7-all.jar/com/puppycrawl/tools/checkstyle/CheckStyleTask.class)

Nuance it that : 
as we move CheckStyleTask.class to "ant" subfolder it will be weird to keep config in parent package.
Example:
checkstyle-6.7-all.jar/com/puppycrawl/tools/checkstyle/ant/CheckStyleTask.class
checkstyle-6.7-all.jar/com/puppycrawl/tools/checkstyle/antlib.xml

Ideal (and clear for users) will be to have:
checkstyle-6.7-all.jar/com/puppycrawl/tools/checkstyle/ant/CheckStyleTask.class
checkstyle-6.7-all.jar/com/puppycrawl/tools/checkstyle/ant/antlib.xml

But in this case we brake compatibility, and as we do that lets rename checkstyletask.properties from first level citizen(checkstyle-6.7-all.jar/checkstyletask.properties) to 
checkstyle-6.7-all.jar/com/puppycrawl/tools/checkstyle/ant/checkstyle-ant-task.properties

in that case root of our checkstyle-6.7-all.jar will be pure checkstyle library content (without any extension). Braking compatibility is not that painful if we provide instructions - straight forward search and replace.

Does it make sense to you ? Do you agree with me ?

File: src/main/resources/checkstyletask.properties [hunk 1]
```diff
@@ -1 +0,0 @@
-checkstyle=com.puppycrawl.tools.checkstyle.CheckStyleTask
```
Comment (mkordas) [inline src/main/resources/checkstyletask.properties]: even `git mv` doesn't preserve history here, I'm not sure why

**Comment by mkordas** [comment]
Yes, I agree with you. Now it makes sense.

Please review the changes. If they are OK, then I'll write migration guide and I'll raise PR for checkstyle-samples.

File: src/main/resources/checkstyletask.properties (hunk 1 above)
Comment (romani) [inline src/main/resources/checkstyletask.properties]: hmm, very strange, try one more time, if still a problem I will apply PR as is (unfortunate , but not blocking issue)

Comment (mkordas) [inline src/main/resources/checkstyletask.properties]: From http://sourceware.org/ml/frysk/2008-q1/msg00004.html:
> git-mv does not do anything special, in fact.  GIT's support for file
renaming is just implicit detection, not explicit markers from user action.
So, what matters is doing a commit that removes file A and adds file B,
where A and B have completely identical contents (matching sha1sum).
To preserve history, you have to carefully do a commit that only renames,
and then a commit that touches the contents of the file.

So I need to make two commits

Comment (romani) [inline src/main/resources/checkstyletask.properties]: please do

**Comment by romani** [comment]
merged as FF

**Comment by romani** [comment]
please write instructions to user to migrate to 6.8 and close original issue.

**Comment by mkordas** [comment]
@romani, you've merged not the latest commit. f9fb945 is ok, but the next one should be 3efd1cf, not e5eb54210a77df36a76713c315a9584bdd8a501b.

**Comment by romani** [comment]
reverted, you commit is applied as cherry-pick.

I always merge by branches: ```./fast-forward-merge.sh checkstyle  mkordas checkstyle_ant_task```
It might be that I did a merge to early, and you was in a middle of push.

**Comment by mkordas** [comment]
Yeah, this was last minute fix - I was experimenting too much with this rename and move. Such changes are very error prone - we should think about integration tests for them. Right now build is always green :)

**Comment by romani** [comment]
While you are in a context of this problem, please open issue with description what commit should be reverted and what error you expect.

**Comment by mkordas** [comment]
Right now `master` is OK, nothing needs to be reverted. I'll create issue to test CheckstyleAntTask in the way it's described in docs. I'll also provide migration guide.

**Pull request closed without merging**